conjure-generator-common/src/main/java/com/palantir/conjure/SimplifiedPattern.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure;
+
+import com.google.errorprone.annotations.Immutable;
+
+/**
+ * A simplified/restricted pattern matcher. Goal is to execute faster than general regex at the cost of reduced
+ * features.
+ *
+ * JMH benchmarks suggest a ~40x increased throughput using the manual coded versions.
+ */
+@Immutable
```

Comment:
Interfaces cannot constrain mutability, nor can the iface itself be mutable
```suggestion
```

Reply:
We actually do need it due to the recursive checking of immutability (unless there's another trick to this)
```
/Volumes/git/conjure/conjure-generator-common/src/main/java/com/palantir/conjure/CaseConverter.java:74: warning: [ImmutableEnumChecker] enums should be immutable: 'Case' has field 'pattern' of type 'com.palantir.conjure.SimplifiedPattern', the declaration of type 'com.palantir.conjure.SimplifiedPattern' is not annotated with @com.google.errorprone.annotations.Immutable
        private final SimplifiedPattern pattern;
                                        ^
    (see https://errorprone.info/bugpattern/ImmutableEnumChecker)
error: warnings found and -Werror specified
1 error
1 warning
```